### PR TITLE
Reader: Fix polling for updates on Discover, blog streams.

### PR DESCRIPTION
**Pull request by blowery**
Some streams don't hand back a date_range, so the code to set the pendingDateAfter was breaking. This changes it to pull the date off of the post in the post store instead.

To test, change `getNextPageParams` in `lib/feed-stream-store/actions.js` to look like this:

```js
function getNextPageParams( store ) {
	var params = {
			orderBy: store.orderBy,
			number: store.perPage,
			meta: 'post,discover_original_post'
		},
		lastDate = store.getLastItemWithDate();

	if ( lastDate ) {
		params.before = lastDate;
	} else {
		// only fetch four items for the initial page
		// speeds up the initial fetch a fair bit
		params.number = 4;
		params.before = '2015-12-01T00:00:00Z'; // important part
	}

	return params;
}
```

and then pull up a site stream, like http://calypso.localhost:3000/read/blog/id/64089429 (Fortune). In prod, this throws an error on the second poll that finds new posts. In dev it just works.

### CODE REVIEW

**Comment by blowery** [comment]
cc @mjangda @bluefuton

File: client/lib/feed-stream-store/actions.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ function getNextPageParams( store ) {
 	} else {
 		// only fetch four items for the initial page
 		// speeds up the initial fetch a fair bit
-		params.number = 4;
+		params.before = '2015-12-01T00:00:00Z';
```
Comment (bluefuton) [inline client/lib/feed-stream-store/actions.js]: Is this left over from testing? `params.number` has gone missing too :)

File: client/lib/feed-stream-store/feed-stream.js [hunk 2]
```diff
@@ -441,7 +441,7 @@ assign( FeedStream.prototype, {
 			postKeys = this.filterNewPosts( data.posts );
 			if ( postKeys.length > 0 ) {
 				this.pendingPostKeys = postKeys;
-				this.pendingDateAfter = moment( data.date_range.after );
+				this.pendingDateAfter = moment( FeedPostStore.get( postKeys[ postKeys.length - 1 ] ).date );
```
Comment (bluefuton) [inline client/lib/feed-stream-store/feed-stream.js]: Can `FeedPostStore.get( postKeys[ postKeys.length - 1 ]` be null in some cases? The tests are complaining `TypeError: Cannot read property 'date' of undefined`: https://circleci.com/gh/Automattic/wp-calypso/3900

Comment (bluefuton) [inline client/lib/feed-stream-store/feed-stream.js]: Ah, I see you've fixed the test stub in 13baa3c :+1:

**Comment by bluefuton** [comment]
Confirmed the bug on wpcalypso: it previously tripped up on `'Uncaught TypeError: Cannot read property 'after' of undefined'` on the second poll.

This no longer happens on this branch :+1:

File: client/lib/feed-stream-store/actions.js (hunk 1 above)
Comment (blowery) [inline client/lib/feed-stream-store/actions.js]: bah!

**Pull request merged**